src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Utf8JsonReader.SingleSegment.Tests.cs

```diff
@@ -0,0 +1,132 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using MicroBenchmarks.Serializers;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public partial class Utf8JsonReaderTests
+    {
+        private const int Iterations = 1_000_000;
+
+        private byte[] s_singleEmptyMultiLineComment;
+        private byte[] s_multipleEmptyMultiLineComment;
+        private byte[] s_singleShortMultiLineComment;
+        private byte[] s_multipleShortMultiLineComment;
+        private byte[] s_singleLongMultiLineComment1;
+        private byte[] s_singleLongMultiLineComment2;
+
+        private byte[] s_singleEmptySingleLineComment;
+        private byte[] s_multipleEmptySingleLineComment;
+        private byte[] s_singleShortSingleLineComment;
+        private byte[] s_multipleShortSingleLineComment;
+        private byte[] s_singleLongSingleLineComment;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            // every test case ends with {} so that we can re-use it for both Skip and Allow options
+            s_singleEmptyMultiLineComment = Encoding.UTF8.GetBytes("/**/{}");
+            s_multipleEmptyMultiLineComment = Encoding.UTF8.GetBytes("/**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**/{}");
```

Comment:
Is it necessary to test both single-line and multi-line comments? Shouldn't the perf characteristics of both generally be the same?

Reply:
no, multi-segment has a bit more checks - it's essentially different code

Reply:
> no, multi-segment has a bit more checks and logic - it's essentially different code

I was asking about single-line and multi-line **comments**, not multi-segment. Does your statement still hold true?

Reply:
@ahsonkhan still different code - it's possible to regress one but not another